### ISSUE TITLE
Adding new field to an existing custom type gives "fieldDefinitions has no changes." error

### ISSUE THREAD

**Issue by mucar3019** [labd/terraform-provider-commercetools#285]
Lets say we got a custom type defined as ;

```hcl
resource "commercetools_type" "review_additional_data" {
  key  = "review-additional-data"
  name = {
    en-US = "review additional data type"
    de-DE = "review additional data type"
  }

  resource_type_ids = ["review"]

  field {
    name  = "productGroupingId"
    label = {
      en-US = "Product Grouping Id"
      de-DE = "Product Grouping Id"
    }
    type {
      name = "String"
    }
  }
}
```

When you add a new field for this type like this;

```hcl
resource "commercetools_type" "review_additional_data" {
  key  = "review-additional-data"
  name = {
    en-US = "review additional data type"
    de-DE = "review additional data type"
  }

  resource_type_ids = ["review"]

  field {
    name  = "productGroupingId"
    label = {
      en-US = "Product Grouping Id"
      de-DE = "Product Grouping Id"
    }
    type {
      name = "String"
    }
  }

  field {
    name  = "isApproved"
    label = {
      en-US = "Approved"
      de-DE = "Approved"
    }
    type {
      name = "Boolean"
    }
  }
}
```
planning results with 1 Change;

```console
# commercetools_type.review_additional_data will be updated in-place
  ~ resource "commercetools_type" "review_additional_data" {
        id                = "d0946f3a-bfee-496d-9fd2-9e9bd6b34196"
        name              = {
            "de-DE" = "review additional data type"
            "en-US" = "review additional data type"
        }
        # (3 unchanged attributes hidden)

      + field {
          + input_hint = "SingleLine"
          + label      = {
              + "de-DE" = "Approved"
              + "en-US" = "Approved"
            }
          + name       = "isApproved"
          + required   = false

          + type {
              + name = "Boolean"
            }
        }
        # (1 unchanged block hidden)
    }

Plan: 0 to add, 1 to change, 0 to destroy.
```

But when you apply it, it returns this error;

```console
commercetools_type.review_additional_data: Modifying... [id=d0946f3a-bfee-496d-9fd2-9e9bd6b34194]
╷
│ Error: 'fieldDefinitions' has no changes.
│
│   with commercetools_type.review_additional_data,
│   on ct-types.tf line 741, in resource "commercetools_type" "review_additional_data":
│  741: resource "commercetools_type" "review_additional_data" {
│
```

This happens on every custom types we have.

**Comment by mvantellingen** [comment]
Can you try version 1.3.0? It seems to work in that release (we rewrote the `type` and `product_type` resources to fix numerous issues)

1.3.0 is being released now, it's the same as v1.3.0-pre.0

**Comment by mvantellingen** [comment]
Closing for now since this should work; feel free to re-open if it doesn't work in 1.3
